File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [0, 0];
    for(let i = 1; i <= Math.sqrt(yellow); i++){
        if(yellow % i === 0) {
            let width = i;
            let height = yellow/i;
            if((width+2) * (height + 2) == brown + yellow) {
                answer[0] = Math.max(width+2, height+2);
                answer[1] = Math.min(width+2, height+2);
            }
        } 
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math 
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    for i in range(3, int(math.sqrt(total))+1) :
        if total % i == 0:
            p = total // i
        else :
            continue
        if (p - 2) * (i - 2) == yellow:
            answer.append(p)
            answer.append(i)
            break
    answer.sort(reverse=True)
    return answer